src/_librsyncmodule.c

```diff
@@ -389,9 +389,21 @@ _librsync_new_patchmaker(PyObject* self, PyObject* args)
   pm->x_attr = NULL;
 
   pm->basis_file = python_file;
-  cfile = fdopen(python_fd, "rb"); /* same mode as in the Python code */
-  if (cfile == NULL) return NULL;
-  pm->patch_job = rs_patch_begin(rs_file_copy_cb, cfile);
+  int my_fd = dup(python_fd);
+  if (my_fd < 0) {
+      char error_string[200];
+      snprintf(error_string, 200, "fdup(): %s (%d)", strerror(errno), errno);
+      PyErr_SetString(librsyncError, error_string);
+      return NULL;
+  }
+  pm->patch_file = fdopen(my_fd, "rb"); /* same mode as in the Python code */
+  if (pm->patch_file == NULL) {
+      char error_string[200];
+      snprintf(error_string, 200, "fdopen(): %s (%d)", strerror(errno), errno);
```

Comment:
I think it would be better to limit the `strerror(errno)` string in size and keep the overall string at a bigger length, making sure that errno in parentheses is output in all cases.
Same applies also to above snprintf.